### ISSUE TITLE
`define(name, body)` documentation does not align with implementation

### ISSUE THREAD

**Issue by voxik** [rpm-software-management/rpm#2962]
This line is incorrect IMHO:

https://github.com/rpm-software-management/rpm/blob/689f1d8d3ef4fcba4b4d05160cb5063b8ac11310/docs/manual/lua.md?plain=1#L120

It should actually be something like `define(name_body)`. This example actually seems to be correct:

https://github.com/rpm-software-management/rpm/blob/689f1d8d3ef4fcba4b4d05160cb5063b8ac11310/docs/manual/lua.md?plain=1#L128

But in any case, it would make more sense if the heading was correct and the `define` method supported the `name` and `body` arguments. Why it does not work like this?

**Comment by voxik** [comment]
IOW this is what the heading says:
~~~
$ rpm -E "%{lua:rpm.define('foo', '1')}" -E "%foo"
error: Macro %foo has empty body
error: lua script failed: [string "<lua>"]:1: error defining macro
~~~

but this is how it actually works:

~~~
$ rpm -E "%{lua:rpm.define('foo 1')}" -E "%foo"

1
~~~

**Comment by mlschroe** [comment]
It's kinda sad that this strips the leading spaces:
```
 rpm -E "%{lua:macros.define({'foo', ' 1 '})}" -E "x%{foo}x"

>1 <
```

**Comment by pmatilai** [comment]
Indeed, rpm.define() calls rpmDefineMacro() which is takes the macro name and the body as a single line - as you would get in a macro file. It doesn't make much sense as an API.

IIRC Lua's rpm.define() used that API because the other alternative lacked all the error checking back in the day, but things are quite different now, Lua should get a better define macro.

Note that this also works:
`macros['foo'] = 1`

**Comment by voxik** [comment]
For the context, I have stumbled upon this issue playing with #2969 and my initial idea was to assign the whole macro body including the new lines to some macro with some specifically crafted name. But I was not able to achieve that no matter what and resorted to use variables (which is probably good enough variant).

It is my feeling that for whatever reason defining macro in Lua is limited comparing to plain .spec file, not being able to insert new line there. But maybe I have just not figured out the right amount of backslashes or what not.

**Comment by pmatilai** [comment]
This got fixed in the newly added manual in 2f60a10414872311948fead686cd71753676a30a
